src/libraries/Common/src/System/Net/NTAuthentication.Common.cs

```diff
@@ -90,6 +91,34 @@ internal bool IsKerberos
             }
         }
 
+        internal bool IsNTLM
+        {
+            get
+            {
+                if (_lastProtocolName == null)
+                {
+                    _lastProtocolName = ProtocolName;
+                }
+
+                return (object)_lastProtocolName == (object)NegotiationInfoClass.NTLM;
```

Comment:
Using reference equality here feels fragile.  We're guaranteed that all of the strings involved will only ever be the same instances?

Reply:
Good catch. This was just moved around between the partial files for the old internal API so it doesn't really break anything but we should definitely change it.

Reply:
I changed it not to use reference equality. This is just internal property that's not exposed by any external API.